posts/postRouter.js

```diff
@@ -1,27 +1,84 @@
-const express = 'express';
+const express = require("express");
 
-const router = express.Router();
+const PostDb = require("./postDb");
 
-router.get('/', (req, res) => {
+const router = express.Router();
 
+// GET POSTS - works
+router.get("/", async (req, res) => {
+  try {
+    const posts = await PostDb.get(req.query);
+    res.status(200).json(posts);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ message: "Server error retrieving posts" });
+  }
 });
 
-router.get('/:id', (req, res) => {
-
+// GET POST BY ID - works
+router.get("/:id", validatePostId, async (req, res) => {
+  try {
+    const post = await PostDb.getById(req.params.id);
```

Comment:
do a check for the get by Id if it exists; if not send a `404`.
If you want to know what it returns just console log post. If it returns an array, then do if `post.length < 0` or if it returns an object you can do if `post` is true or not.